qiskit_experiments/database_service/db_analysis_result.py

```diff
@@ -161,19 +151,39 @@ def save(self) -> None:
                 "Analysis result cannot be saved because no experiment service is available."
             )
             return
-
-        _result_data = json.loads(json.dumps(self._result_data, cls=self._json_encoder))
-        _result_data["_source"] = self._source
+        result_data = {}
+
+        # Temp hack for FitVal
+        if isinstance(self.value, FitVal):
+            value = json.loads(json.dumps(self.value, cls=self._json_encoder))
+            result_data["__type__"] = value["__type__"]
+            result_data["_value"] = value["args"][0]
+            if self.value.stderr is not None:
+                result_data["_stderr"] = value["args"][1]
+            if self.value.unit is not None:
+                result_data["_unit"] = value["args"][2]
```

Comment:
Yuck

Reply:
Why do you need to take the round trip through `json.dumps` -> `json.loads`? This would be much easier to read if it directly referenced the field names of the `FitVal` instead of `value["args"][1]`, etc.

Reply:
Or why not move this manipulation to `ExperimentEncoder`?

Reply:
This was something @jyu00 had set up in the original code. I. think it has something to do with how service code did json serializer that needed updating to be able to pass a custom encoder.